source/server/listener_manager_impl.h

```diff
@@ -390,13 +390,15 @@ class ListenerImpl : public Network::ListenerConfig,
 class ListenerFilterChainFactoryBuilder : public FilterChainFactoryBuilder {
 public:
   ListenerFilterChainFactoryBuilder(
-      ListenerImpl& listener, Configuration::TransportSocketFactoryContextImpl& factory_context);
+      ListenerImpl& listener, Configuration::TransportSocketFactoryContextImpl& factory_context,
+      bool is_quic);
```

Comment:
My impression is that TCP`ListenerFilterChainFactoryBuilder` and QUIC one has nothing shared... Why not derived QUIC one from FilterChainFactoryBuilder?

Reply:
both of them access ListenerImpl and TransportSocketFactoryContextImpl. I made QuicListenerFilterChainFactoryBuilder a stand alone class deriving from ListenerFilterChainFactoryBuilder.